lib/search_logic.rb

```diff
@@ -0,0 +1,131 @@
+module SearchLogic
+    SEARCH_LIMIT = 20
+
+    class << self
+        def get_places_from_city_string(city_name, place_type)
+            category = case place_type
+            when "attractions"
+              "tourism.attraction"
+            when "hotels"
+              "accommodation.hotel"
+            when "restaurants"
+              "catering.restaurant"
+            else
+              raise PlaceTypeInvalid.new("Error: The place type used is not supported at the moment")
+            end
+
+            city = get_city_from_name(city_name)
+            if city.public_send(place_type).count == SEARCH_LIMIT
+                return city.public_send(place_type)
+            else
+                response = get_place_details(category, city.city_place_id)
+                send("create_#{place_type}", response, city)
+            end
```

Comment:
By calling `.count` and then iterating through the collection later, you actually make two database queries instead of one, so the "old way" was better.

```suggestion
            city = get_city_from_name(city_name)
            places = city.public_send(place_type).limit(SEARCH_LIMIT).to_a
            return if places.count >= SEARCH_LIMIT

            response = get_place_details(category, city.city_place_id)
            send("create_#{place_type}", response, city)
```